### PR TITLE
Basic unit tests for arm generate

### PR DESCRIPTION
**Pull request by asalkeld**

### CODE REVIEW

File: pkg/arm/arm_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		wantErr           bool
+		wantResourceTypes []string
+	}{
+		{
+			name:     "good-create",
+			cs:       fixtures.NewTestOpenShiftCluster(),
+			isUpdate: false,
+			wantErr:  false,
+			wantResourceTypes: []string{
+				"\"type\": \"Microsoft.OperationalInsights/workspaces\"",
```
Comment (jim-minter) [inline pkg/arm/arm_test.go]: use backticks here to avoid the backslashes

File: pkg/arm/arm_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"strings"
+	"testing"
+
+	acsapi "github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/config"
+	"github.com/openshift/openshift-azure/pkg/util/fixtures"
+	"github.com/sirupsen/logrus"
+)
+
+func Test_simpleGenerator_Generate(t *testing.T) {
```
Comment (jim-minter) [inline pkg/arm/arm_test.go]: no underscores please

File: pkg/arm/arm_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	acsapi "github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/config"
+	"github.com/openshift/openshift-azure/pkg/util/fixtures"
+	"github.com/sirupsen/logrus"
```
Comment (kargakis) [inline pkg/arm/arm_test.go]: nit: group imports as discussed elsewhere

Comment (jim-minter) [reply pkg/arm/arm_test.go]: @kargakis I feel like if you really want this enforced, a lint tool is in order.  I don't think it's worth the effort.

**Pull request closed without merging**